source/extensions/filters/http/rbac/rbac_filter.cc

```diff
@@ -44,7 +44,16 @@ Http::FilterHeadersStatus RoleBasedAccessControlFilter::decodeHeaders(Http::Head
   const Filters::Common::RBAC::RoleBasedAccessControlEngine& engine =
       config_->engine(callbacks_->route());
 
-  if (engine.allowed(*callbacks_->connection(), headers)) {
+  bool darklaunch_allowed;
+  bool enforcement_allowed = engine.allowed(*callbacks_->connection(), headers, darklaunch_allowed);
+
+  if (darklaunch_allowed) {
+    config_->stats().darklaunch_allowed_.inc();
+  } else {
+    config_->stats().darklaunch_denied_.inc();
```

Comment:
For both the existing stats as well as these dark ones, I feel like we should apply tags with the matching policy name so we can identify the source of the allow/deny.

Reply:
agree, will put this in separated PR about what attributes need to be collected and how to pass data to metrics storage(pass data cross-filters, etc)